test/src/compose/application-manager.spec.ts

```diff
@@ -389,6 +389,110 @@ describe('compose/application-manager', () => {
 			.that.deep.includes({ serviceName: 'main' });
 	});
 
+	it('infers a kill step when a service has to be updated but the strategy is delete-then-download', async () => {
```

Comment:
I don't think this level of unit testing could not have been sanely done if it weren't for all the testing improvements we've made to the Supervisor...this is super cool to see.

Reply:
I think it could still be made simpler, there is a lot of boilerplate needed now, but is a better situation than what we had before for sure.